File: JULY-2025/28-07-2025/p2044.scala
object Solution {
    def countMaxOrSubsets(nums: Array[Int]): Int = {
        val maxOr = nums.reduce(_ | _)
        def dfs(i: Int, orVal: Int): Int = {
            if (i == nums.length) if (orVal == maxOr) 1 else 0
            else dfs(i + 1, orVal | nums(i)) + dfs(i + 1, orVal)
        }
        dfs(0, 0)
    }
}

File: JULY-2025/28-07-2025/p2044.rs
impl Solution {
    pub fn count_max_or_subsets(nums: Vec<i32>) -> i32 {
        let max = nums.iter().fold(0, |acc, &x| acc | x);
        let mut result = 0;
        fn dfs(nums: &[i32], idx: usize, cur: i32, max: i32, result: &mut i32) {
            if idx == nums.len() {
                if cur == max {
                    *result += 1;
                }
                return;
            }

            dfs(nums, idx + 1, cur | nums[idx], max, result);
            dfs(nums, idx + 1, cur, max, result);
        }
        dfs(&nums[..], 0, 0, max, &mut result);
        result
    }
}